src/Filters/AbstractAuthFilter.php

```diff
@@ -7,14 +7,16 @@
 use CodeIgniter\Filters\FilterInterface;
 use CodeIgniter\HTTP\RedirectResponse;
 use CodeIgniter\HTTP\RequestInterface;
-use CodeIgniter\HTTP\Response;
 use CodeIgniter\HTTP\ResponseInterface;
+use CodeIgniter\Shield\Config\Auth;
 
 /**
  * Group Authorization Filter.
  */
 abstract class AbstractAuthFilter implements FilterInterface
 {
+    protected string $filterName;
```

Comment:
@samsonasik  Thanks for the review, honestly, it blew my mind.

This implementation can only be used for `group` and `permission` (It is available in the shield by default) according to the defined `switch`.
 It effectively renders inheritance from extend useless.
It seems that it should be marked as `@internal`.

In this regard, I am not sure which is the better way.

Reply:
I never use this library, so let see other opinions ;)

Reply:
@datamweb This change for `AbstractAuthFilter` is simply bad design.
Because abstraction should not know details (concrete classes).

The `AbstractAuthFilter` is the base class. Devs can extend it and create their own filters. But the change makes it impossible.

I think the default two filters is not enough for all needs. Someone may need a filter to check both group and permission.

Reply:
@Kenjis I changed the implementation, I think now there is no problem to create a custom filter that inherits from `AbstractAuthFilter`.

Reply:
Much better!

But this is still a breaking change. The `redirectToDeniedUrl()` method must be added if a dev extends `AbstractAuthFilter`.

Do we write to documentation for upgrade?
https://github.com/codeigniter4/shield/blob/develop/UPGRADING.md
Or if we add the default implementation (redirect to `/`)  in `AbstractAuthFilter`, it will not be a breaking change.